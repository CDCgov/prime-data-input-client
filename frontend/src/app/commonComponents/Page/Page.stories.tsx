import { Story, Meta } from "@storybook/react";

import Page from "./Page";

export default {
  title: "Components/Page",
  component: Page,
  argTypes: {},
} as Meta;

const Template: Story = () => <Page />;

export const WithChildren = Template.bind({});
WithChildren.args = {
  children: (
    <>
      <p>This is some test content</p>
    </>
  ),
};
