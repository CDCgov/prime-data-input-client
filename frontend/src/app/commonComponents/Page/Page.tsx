import React from "react";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

import USAGovBanner from "../USAGovBanner";

const Page: React.FC<{}> = ({ children }) => (
  <div className="App">
    <div id="main-wrapper">
      <USAGovBanner />
      {children}
      <ToastContainer
        autoClose={5000}
        closeButton={false}
        limit={2}
        position="bottom-center"
        hideProgressBar={true}
      />
    </div>
  </div>
);

export default Page;
