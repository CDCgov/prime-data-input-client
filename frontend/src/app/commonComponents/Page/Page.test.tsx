import { render } from "@testing-library/react";

import Page from "./Page";

describe("Page", () => {
  describe("No children", () => {
    it("renders", () => {
      expect(<Page />).toMatchSnapshot();
    });
  });
  describe("With children", () => {
    it("displays children", () => {
      const { getByText } = render(<Page>Hello World</Page>);
      expect(getByText("Hello World")).toBeInTheDocument();
    });
  });
});
