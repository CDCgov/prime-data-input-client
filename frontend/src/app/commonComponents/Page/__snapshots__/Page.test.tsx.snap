// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page No children renders 1`] = `<Page />`;
